### PR TITLE
interactive verifier example

### PR DESCRIPTION
**Pull request by heeckhau**
Example for the interactive verifier.

To the reviewers: please check the code, structure and comments. I expect people to copy this code and use it as a template for getting started.

### CODE REVIEW

File: notary-server/config/config.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ server:
   port: 7047
 
 notarization:
-  max-transcript-size: 16384
+  max-transcript-size: 20480
```
Comment (yuroitaki) [inline notary-server/config/config.yaml:7]: modified this to match the latest default (recv + sent) used in prover

**Comment by yuroitaki** [comment]
Added commits to fix twitter and discord examples with changes at notary server config (and formatting fix of 2 component files as a result of running `cargo fmt`)

File: notary-server/openapi.yaml [hunk 2]
```diff
@@ -175,8 +175,11 @@ components:
           enum:
             - "Tcp"
             - "Websocket"
-        maxTranscriptSize:  
-          description: Maximum transcript size in bytes
+        maxSentData:  
+          description: Maximum data that can be sent by the prover in bits
```
Comment (themighty1) [inline notary-server/openapi.yaml:179]: shouldnt this be bytes, not bits.

btw, where in the code does the server handle `maxSentData`? Grepping this crate didnt find where it is used.

File: tlsn/examples/interactive/interactive.rs [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+
+#[tokio::main]
+async fn main() {
+    tracing_subscriber::fmt::init();
+
+    let uri = "https://notary.pse.dev/info";
+    let id = "interactive verifier demo";
+
+    // connect prover and verifier
+    let (prover_socket, verifier_socket) = tokio::io::duplex(2 << 23);
```
Comment (themighty1) [inline tlsn/examples/interactive/interactive.rs:26]: usually we write 1 << 23

File: tlsn/examples/interactive/interactive.rs [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+        .find("BEGIN PUBLIC KEY")
+        .expect("Expected valid public key in JSON response");
+    // Check Session info: server name
+    assert_eq!(session_info.server_name.as_str(), SERVER_DOMAIN);
+
+    (sent, received, session_info)
+}
+
+/// Redact and reveal received data to the verifier
+fn react_and_reveal_received_data(mut prover: Prover<Prove>) -> Prover<Prove> {
```
Comment (themighty1) [inline tlsn/examples/interactive/interactive.rs:146]: /s/react/redact

File: tlsn/examples/interactive/interactive.rs [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+    json.public_key
+        .find("BEGIN PUBLIC KEY")
+        .expect("Expected valid public key in JSON response");
+    // Check Session info: server name
+    assert_eq!(session_info.server_name.as_str(), SERVER_DOMAIN);
+
+    (sent, received, session_info)
+}
+
+/// Redact and reveal received data to the verifier
```
Comment (themighty1) [inline tlsn/examples/interactive/interactive.rs:145]: acc to the style guide method comments should be formatted as:
```suggestion
/// Redacts and reveals received data to the verifier.
```

File: notary-server/openapi.yaml (hunk 2 above)
Comment (yuroitaki) [reply notary-server/openapi.yaml:179]: https://github.com/tlsnotary/tlsn/blob/dev/notary-server/src/domain/notary.rs#L24-L26 — just different casing type used, we convert snake_case used internally to camelCase in payload for clients (which I supposed is more standard)

File: tlsn/examples/interactive/interactive.rs (hunk 4 above)
Comment (th4s) [inline tlsn/examples/interactive/interactive.rs:146]: Do we have to consume the `Prover` here? Usually it is better to do
`fn redact_and_reveal_received_data(prover: &mut Prover<Prove>) {...}`

File: tlsn/examples/interactive/interactive.rs [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+    _ = prover.reveal(0..commit_hash_start, Direction::Received);
+    _ = prover.reveal(
+        commit_hash_start + commit_hash.len()..recv_transcript_len,
+        Direction::Received,
+    );
+    prover
+}
+
+/// Redact and reveal sent data to the verifier
+fn react_and_reveal_sent_data(mut prover: Prover<Prove>) -> Prover<Prove> {
```
Comment (th4s) [inline tlsn/examples/interactive/interactive.rs:170]: same

**Review by th4s — APPROVED** [review]
Nice :+1:

**Review by themighty1 — APPROVED** [review]

**Pull request merged**